blog/04-02-2023-build-a-donation-dApp-on-celo-to-award-your-favourite-content-creator/04-02-2023-build-a-donation-dApp-on-celo-to-award-your-favourite-content-creator.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Build a Donation dApp on Celo to award your Favorite Content Creator
+description: Build a Donation dApp on Celo to award your Favorite Content Creator
+authors:
+  - name: Aviraj Khare
+url: https://github.com/avirajkhare00
+image_url: https://github.com/avirajkhare00.png
+tags: [celosage, advanced]
+hide_table_of_contents: false
+slug: /tutorials/build-a-donation-dapp-on-celo-to-award-your-favourite-content-creator
+---
+
+![header](../../src/data-tutorials/showcase/advanced/build-a-donation-dapp-on-celo-to-award-your-favourite-content-creator.png)
+
+## Introduction
+
+Hello guys. Today we are going to build a donation dApp. Building a donation dApp is an excellent example of a Blockchain use case. Building a donation dApp (decentralized application) on the blockchain can provide several benefits over traditional donation platforms. These include increased transparency, as all transactions on the blockchain are recorded on a public ledger that anyone can audit.
+
+Additionally, using a blockchain-based platform can increase the security of donations, as it is more difficult to hack or tamper with transactions on a decentralized network. Additionally, it can increase trust and transparency in the donation process. It will be less prone to fraud or scams as the transactions are recorded on a public ledger and can be audited by anyone.
+
+## Prerequisites
+
+ - Knowledge of Solidity.
+ - Knowledge of HTML, CSS, JavaScript.
+
+## Requirements
+
+ - **Hardhat:** We will use hardhat to write smart contract.
+ - **Code editor:** Code editor of your choice.
+
+## Let's go 🔥🔥🔥
+
+Here is a list of what we are going to cover in this tutorial:
+ - **Creating smart contract using hardhat.**
+ - **Deploying smart contract on CELO testnet using hardhat.**
+ - **Creating frontend using vanilla HTML, CSS and JavaScript.**
+
+To begin with, we need to have a clear picture what our dApp is going to do. We want to donate to our favourite content creators. Here is how it is going to work:
+
+1. Creator can get onboarded using our UI by uploading any digital file. This file will be stored in IPFS from the frontend.
+2. Anyone will be able to reward the creator expect creator itself.
+3. When user reward the creator in CELO, the amount will go directly to the creator.
+
+Now we have the clear picture, let's get started...
+
+## Code
+
+### Smart Contract
+
+Below is the smart contract that will faciliate all the features.
+
+```js
```

Comment:
Change ``` js  to solidity ```

Reply:
OK

Reply:
Markdown doesn't detect `solidity` so it's `js`

Reply:
In all my blogs it's `js`

Reply:
@avirajkhare00 it does detect solidity you can check my published posts. https://github.com/celo-org/docs/pull/584